### PR TITLE
Feature/lowpass filter

### LINKED ISSUE #173: Develop stable low pass filter algorithms for voltages and temperatures
### Current Features

We have used various instable, volatile, and aggressive algorithms to eliminate noise form our data readings. These are case-specific to behavior weve seen, not strong agaisnt edge cases, and generally over aggresive

### Desired Additional Features

We should create stable low pass filtering to eliminate high frequency noise. This should be a simple and generalized filter that can be applied to data of any kind. It should not be especially aggressive and should just work as a no brainer to apply to any incoming data 

### Screenshots (as needed)

_No response_

### PR DESCRIPTION
**Pull request by DimitryP6**
## Changes

Added the simple_ema.h file and renamed branch

## Notes

Alpha works well for 0.1

## Test Cases

- Case A
- Edge case
- ...

## To Do

_Any remaining things that need to get done_

- [ ] item 1
- [ ] ...

## Checklist

It can be helpful to check the `Checks` and `Files changed` tabs.
Please reach out to your Project Lead if anything is unclear.
Please request reviewers and ping on slack only after you've gone through this whole checklist.

- [ *] All commits are tagged with the ticket number
- [* ] No merge conflicts
- [ *] All checks passing
- [* ] Remove any non-applicable sections of this template
- [* ] Assign the PR to yourself
- [* ] Request reviewers & ping on Slack
- [* ] PR is linked to the ticket (fill in the closes line below)

Closes #173

### CODE REVIEW

File: middleware/simple_ema.h [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#ifndef SIMPLE_EMA
+#define SIMPLE_EMA
+
+// Exponential Moving Average Lowpass Filter
+ema_filter(float current_value, float *previous_ema, float alpha);
```
Comment (jr1221) [inline middleware/simple_ema.h:5]: needs void no?

A good way to ensure your code builds is to run it in Cerberus.
Steps to do so:
1. Commit to ur branch in embedded base
2. Clone `Cerberus`
3. Enter the ./Drivers/Embedded-Base directory in Cerberus and run `git fetch` and `git checkout <your-branch-name>`
4. Now update cerberus `Makefile` with the relevant .c file you want to test
5. run `ner build` and observe any errors/warnings

Lmk if you have questions about these steps, if you need instructions to setup `ner build` check [here](https://nerdocs.atlassian.net/wiki/spaces/NER/pages/524451844/2024+Firmware+Onboarding+Master)

**Review by jr1221 — CHANGES_REQUESTED** [review]
see comment

**Review by jr1221 — APPROVED** [review]
lgtm
